ddtrace/tracer/option.go

```diff
@@ -871,34 +872,51 @@ type UserMonitoringOption func(Span)
 // WithUserEmail returns the option setting the email of the authenticated user.
 func WithUserEmail(email string) UserMonitoringOption {
 	return func(s Span) {
-		s.SetTag("usr.email", email)
+		s.SetTag(ext.UserEmail, email)
 	}
 }
 
 // WithUserName returns the option setting the name of the authenticated user.
 func WithUserName(name string) UserMonitoringOption {
 	return func(s Span) {
-		s.SetTag("usr.name", name)
+		s.SetTag(ext.UserName, name)
 	}
 }
 
 // WithUserSessionID returns the option setting the session ID of the authenticated user.
 func WithUserSessionID(sessionID string) UserMonitoringOption {
 	return func(s Span) {
-		s.SetTag("usr.session_id", sessionID)
+		s.SetTag(ext.UserSessionID, sessionID)
 	}
 }
 
 // WithUserRole returns the option setting the role of the authenticated user.
 func WithUserRole(role string) UserMonitoringOption {
 	return func(s Span) {
-		s.SetTag("usr.role", role)
+		s.SetTag(ext.UserRole, role)
 	}
 }
 
 // WithUserScope returns the option setting the scope (authorizations) of the authenticated user.
 func WithUserScope(scope string) UserMonitoringOption {
 	return func(s Span) {
-		s.SetTag("usr.scope", scope)
+		s.SetTag(ext.UserScope, scope)
+	}
+}
+
+// WithPropagation returns the option allowing the user id to be propagated through distributed traces.
+// The user id is base64 encoded and added to the datadog propagated tags header.
+// RFC https://docs.google.com/document/d/1T3qAE5nol18psOaHESQ3r-WRiZWss9nyGmroShug8ao
```

Comment:
Is this document publicly viewable? If not, is there some public facing documentation that can be linked to instead, or can we expand on the upside/downsides of using this feature?

Alternatively if that kind of documentation isn't ready can we put an explicit comment / note on this that this function is NOT recommended for use?

Reply:
The public documentation for `SetUser()` is here, which in turns links to the go doc where the options are described: https://docs.datadoghq.com/security_platform/application_security/setup_and_configure/?tab=set_user#add-user-information-to-traces.
I'll rework the comments so that the `SetUser()` comment links to this doc and mentions the use case for `WithPropagation()`.
I'll also update the option's comment to warn about sensitive data leak.

Reply:
6883df43b7ee1